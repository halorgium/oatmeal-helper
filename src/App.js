import React from 'react'
import Timer from './Timer'
import Controls from './Controls'
import AbsoluteControl from './AbsoluteControl'
import DeltaControl from './DeltaControl'

import './App.css'

function App () {
  return (
    <Timer>
      <div className='App'>
        <header className='header'>
          <h1 className='title'>Make Oatmeal</h1>
        </header>
        <div>Time right now: <Timer.Now /></div>
        <Timer.ControlReady initial='06:30'>
          <div>Time to be ready: <Controls.Value /></div>
          <AbsoluteControl time='06:30' />
          <AbsoluteControl time='07:00' />
          <AbsoluteControl time='08:00' />
          <DeltaControl diff='00:15' />
          <DeltaControl diff='01:00' />
        </Timer.ControlReady>
        <Timer.ControlCook initial='02:30'>
          <div>Time to cook: <Controls.Value /></div>
          <AbsoluteControl time='02:30' />
          <AbsoluteControl time='08:00' />
          <DeltaControl diff='00:15' />
          <DeltaControl diff='01:00' />
        </Timer.ControlCook>
        <div>Time to wait: <Timer.Wait /></div>
      </div>
    </Timer>
  )
}

App.propTypes = {}

export default App
