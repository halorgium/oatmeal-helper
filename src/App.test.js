import React from 'react'
import renderer from 'react-test-renderer'
import moment from 'moment'
import MockDate from 'mockdate'
import App from './App'

it('renders without crashing', () => {
  MockDate.set(moment('2017-02-08 21:21:00.000Z'))
  const tree = renderer.create(
    <App />
  ).toJSON()
  MockDate.reset()
  expect(tree).toMatchSnapshot()
})
