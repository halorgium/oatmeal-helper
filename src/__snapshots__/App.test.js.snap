// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Make Oatmeal
    </h1>
  </header>
  <div>
    Time right now: 
    13:21
  </div>
  <div>
    Time to be ready: 
    06:30
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    06:30
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    07:00
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    08:00
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    +
    15
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    -
    15
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    +
    01:00
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    -
    01:00
  </button>
  <div>
    Time to cook: 
    02:30
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    02:30
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    08:00
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    +
    15
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    -
    15
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    +
    01:00
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    -
    01:00
  </button>
  <div>
    Time to wait: 
    14:39
  </div>
</div>
`;
